File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program93.java
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Accept character from user. 
//		If it is capital then display all the characters from the input characters till Z. 
//		If input character is small then print all the characters in reverse order till a.
//		 In other cases return directly.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class Solution
{

	public char ch;

	public Solution(char ch)
	{
		this.ch = ch;
	}

	public void Display()
	{	
		
		if(ch>='A' && ch<='Z')
		{
			while(ch!='[')
			{
				System.out.print(ch+" ");
				ch++;
			}
		}

		else if(ch>='a'&& ch<='z')
		{
			while(ch!='`')
			{
				System.out.print(ch+" ");
				ch--;
			}
		
		}
		System.out.println();
	}


}

class Program
{
	public static void main(String[] args)
	{
		Scanner sObj = new Scanner(System.in);

		System.out.println("Enter character:");
		char ch = sObj.next().charAt(0);

		Solution obj = new Solution(ch);
		obj.Display();
	}
}

/*
OUTPUT:
Enter character:
Q
Q R S T U V W X Y Z
Enter character:
m
m l k j i h g f e d c b a 

*/

File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program93.cpp
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Accept character from user. 
//		If it is capital then display all the characters from the input characters till Z. 
//		If input character is small then print all the characters in reverse order till a.
//		 In other cases return directly.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class Solution
{
	private:
		char ch;
		
	public:
		Solution(char ch='\0')
		{
			this->ch = ch;
		}

		void Display()
		{
			
			if(ch>='a'&& ch<='z')
			{
				while(ch!='`')
				{
					cout<<ch<<" ";
					ch--;
				}

			}

			else if(ch>='A' && ch<='Z')
			{
				while(ch!='[')
				{
					cout<<ch<<" ";
					ch++;
				}
			}

			cout<<"\n";

		}
};

int main()
{
	char ch='\0';

	cout<<"Enter the character: ";
	cin>>ch;

	Solution *dObj = new Solution(ch);
	dObj->Display();
	
	delete dObj;
	return 0;
}

/*
OUTPUT:
Enter the character: Q
Q R S T U V W X Y Z 

Enter the character: m
m l k j i h g f e d c b a 

*/
